### PR TITLE
Restore

### PR DESCRIPTION
**Pull request by INR4GE**

### CODE REVIEW

File: src/configs/index.js [hunk 1]
```diff
@@ -107,16 +107,47 @@ export const EMPTY_FIELD = [
...
+  INPUT_NOT_VALID: [
+    {
+      lang: 'uk',
+      value: 'Ви ввели невірні дані'
+    },
+    {
+      lang: 'eng',
+      value: 'You have typed wrong data'
+    }
+  ]
```
Comment (VolodymyrTrach) [inline src/configs/index.js:130]: Move all translations to translations folder

File: src/pages/new-password/new-password.styles.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    top: '202px',
+    right: '56px',
+    padding: '50px',
+    paddingBottom: '79px'
+  },
+  heading: {
+    fontSize: '24px',
+    lineHeight: '29px',
+    marginBottom: '25px',
+    marginTop: '0px',
```
Comment (VolodymyrTrach) [inline src/pages/new-password/new-password.styles.js:26]: Please use rem or em instead of px

File: src/pages/recovery/recovery.styles.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    position: 'absolute',
+    top: '202px',
+    right: '56px',
+    padding: '50px'
+  },
+  heading: {
+    fontSize: '24px',
+    lineHeight: '29px',
+    marginBottom: '25px',
+    marginTop: '0px',
```
Comment (VolodymyrTrach) [inline src/pages/recovery/recovery.styles.js:25]: Please use rem or em instead of px

File: src/configs/index.js (hunk 1 above)
Comment (INR4GE) [reply src/configs/index.js]: moved

File: src/configs/index.js [hunk 4]
```diff
@@ -253,3 +286,96 @@ export const CONFIRM_ERROR = [
...
+export const NEW_PASSWORD_SUCCESS_MESSAGE = [
+  {
+    h2: 'Ваш пароль успішно змінено!',
+    p: 'Вас буде спрямовано на сторінку логінування'
+  },
+  {
+    h2: 'Your password is successfully changed!',
+    p: 'You will be directed to login page'
+  }
+];
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/configs/index.js]: moved

File: src/configs/index.js [hunk 5]
```diff
@@ -38,51 +38,6 @@ export const LANGUAGES_LIST = [
   { lang: 'EN', value: 1 }
 ];
 
-export const CONFIRM_EMAIL = [
-  {
```
Comment (bandvov) [inline src/configs/index.js:42]: please move translation to translations folder

File: src/pages/register/register.js [hunk 6]
```diff
@@ -59,11 +57,11 @@ function Register() {
     if (allFieldsValidated) {
       try {
         setLoading(true);
-        await registerUser(user);
+        await registerUser(user, language);
```
Comment (Alan18081) [inline src/pages/register/register.js:60]: Don't call async actions in the components as much as possible, please. Move the call of the mutation for user creation to the `user.saga` and here only dispatch an action to trigger an action. In component, we should avoid `awaits` as much as possible

File: package.json [hunk 7]
```diff
@@ -42,7 +42,8 @@
     "redux": "^4.0.5",
     "redux-saga": "^1.1.3",
     "redux-saga-test-plan": "^4.0.0-rc.3",
-    "unfetch": "^4.1.0"
+    "unfetch": "^4.1.0",
+    "webpack": "^4.44.1"
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply package.json]: fixed

File: src/pages/confirmation/confirmUser.js [hunk 8]
```diff
@@ -1,9 +1,9 @@
 import { setItems } from '../../utils/client';
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/confirmation/confirmUser.js]: fixed

File: src/pages/confirmation/confirmation.js [hunk 9]
```diff
@@ -15,13 +19,21 @@ const Confirmation = ({ token }) => {
     language: state.Language.language
   }));
 
+  const dispatch = useDispatch();
+
   useEffect(() => {
     setLoading(true);
-    confirmUser(token, language)
+    confirmUser(token)
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/confirmation/confirmation.js]: fixed

File: src/pages/confirmation/confirmUser.js [hunk 10]
```diff
@@ -1,9 +1,9 @@
...
 
-const confirmUser = (token, language) => {
+const confirmUser = (token) => {
   const query = ` 
   mutation {
-    confirmUser(token: "${token}", language: ${language})
+    confirmUser(token: "${token}")
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/confirmation/confirmUser.js]: fixed

File: src/pages/confirmation/confirmation.js [hunk 11]
```diff
@@ -39,7 +51,14 @@ const Confirmation = ({ token }) => {
             <h3>{WELCOME_MESSAGE[language].h3}</h3>
           </>
         )}
-        <Button variant='contained'>{WELCOME_MESSAGE[language].button}</Button>
+        <div className={styles.buttonGroup}>
+          <Button variant='contained' onClick={() => goTo('/')}>
+            {WELCOME_MESSAGE[language].button_1}
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/confirmation/confirmation.js]: fixed

File: src/pages/new-password/new-password.js [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+  const { language } = useSelector((state) => ({
+    language: state.Language.language
+  }));
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    let isSubscribed = true;
+    setLoading(true);
+    checkIfTokenIsValid(token)
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/new-password/new-password.js]: fixed

File: src/pages/new-password/new-password.js [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+          setLoading(false);
+        }
+      });
+    return () => {
+      isSubscribed = false;
+    };
+    // eslint-disable-next-line
+  }, []);
+
+  // LOADING & ERRORS
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/new-password/new-password.js]: fixed

File: src/pages/new-password/resetPassword.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import { setItems } from '../../utils/client';
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/new-password/resetPassword.js]: fixed

File: src/pages/recovery/recoverUser.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
+import { setItems } from '../../utils/client';
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/recovery/recoverUser.js]: fixed

File: src/utils/checkIfTokenIsValid.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
+import { setItems } from './client';
+
+const checkIfTokenIsValid = (token) => {
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/utils/checkIfTokenIsValid.js]: fixed

File: src/translations/user.translations.js [hunk 17]
```diff
@@ -0,0 +1,346 @@
+export const CONFIRM_EMAIL = [
+  {
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/translations/user.translations.js:2]: fixed

File: src/redux/user/user.sagas.js [hunk 18]
```diff
@@ -6,18 +6,23 @@ import { setItems } from '../../utils/client';
 
 export const loginUser = (payload) => {
   const query = ` 
-  mutation login($user: UserInput!) {
+  mutation {
   loginUser(
-    user: $user
+    loginInput: {
+      email: "${payload.email}"
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/redux/user/user.sagas.js:26]: removed

File: src/pages/register/register.js (hunk 6 above)
Comment (INR4GE) [reply src/pages/register/register.js:60]: removed

Comment (INR4GE) [reply src/pages/register/register.js:60]: removed

File: src/redux/user/user.reducer.js [hunk 19]
```diff
@@ -33,7 +40,37 @@ const userReducer = (state = initialState, action = {}) => {
...
+      ...state,
+      passwordReset: action.payload,
+      error: false,
+      userLoading: false
+    };
+  case SET_USER_REGISTERED:
+    return {
+      ...state,
+      userRegistered: action.payload,
+      error: false,
```
Comment (Alan18081) [inline src/redux/user/user.reducer.js:72]: If `error` contains actual error object, it should have false value of `null`

File: src/redux/user/user.sagas.js [hunk 20]
```diff
@@ -1,36 +1,172 @@
...
+    email
+    firstName
   }
 }
   `;
-  return setItems(query, payload);
+  return setItems(query, data);
+};
+
+export const confirmUser = (token) => {
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:41]: There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there

File: src/redux/user/user.sagas.js [hunk 21]
```diff
@@ -1,36 +1,172 @@
...
+  const query = ` 
+  mutation confirmUser($token: String!){
+    confirmUser(token: $token)
+  }
+  `;
+  return setItems(query, { token });
+};
+
+export const recoverUser = (data) => {
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:50]: There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there

File: src/redux/user/user.sagas.js [hunk 22]
```diff
@@ -1,36 +1,172 @@
...
+  const query = ` 
+  mutation recovery($email: String!, $language: Int!){
+    recoverUser(email: $email, language: $language)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const resetPassword = (data) => {
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:59]: There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there

File: src/redux/user/user.sagas.js [hunk 23]
```diff
@@ -1,36 +1,172 @@
...
+  const query = ` 
+  mutation reset($password: String!, $token: String!){
+    resetPassword(password: $password, token: $token)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const checkIfTokenIsValid = (data) => {
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:68]: There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there

File: src/redux/user/user.sagas.js [hunk 24]
```diff
@@ -1,36 +1,172 @@
...
+  const query = ` 
+  mutation checkToken($token: String!){
+  checkIfTokenIsValid(token: $token)
+  }
+  `;
+  return setItems(query, data);
+};
+
+export const registerUser = (data) => {
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:77]: There is no sense to have it as a separate function because it is going to be used only once - inside the saga, please inline its code there

File: src/redux/user/user.sagas.js [hunk 25]
```diff
@@ -1,36 +1,172 @@
...
+  }
+}
+
+export function* handlePasswordReset({ payload }) {
+  try {
+    yield put(resetState());
+    yield put(setUserLoading(true));
+    yield call(resetPassword, payload);
+    yield put(setPasswordReset(true));
+    yield delay(SHOW_AFTER);
```
Comment (Alan18081) [inline src/redux/user/user.sagas.js:133]: Please, rename `SHOW_AFTER` to something lie `REDIRECT_TIMEOUT`, it is much more suitable name for that stuff

File: src/pages/recovery/recovery.js [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+              label={RECOVERY_MESSAGES[language].label}
+              className={`${styles.emailInput} ${
+                !error ? styles.helperEmail : null
+              }`}
+              fullWidth
+              variant='outlined'
+              type='text'
+              helperText={
+                shouldValidate && !emailValidated && email
+                  ? errorMessages[language].value.email
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/recovery/recovery.js:87]: Just following the requirements

File: src/pages/register/register.js [hunk 27]
```diff
@@ -27,21 +25,17 @@ function Register() {
   const [emailValidated, setEmailValidated] = useState(false);
   const [passwordValidated, setPasswordValidated] = useState(false);
   const [allFieldsValidated, setAllFieldsValidated] = useState(false);
-  const [isConfirmedPassword, setIsConfirmedPassword] = useState(false);
+  const [isConfirmedPassword, setIsConfirmedPassword] = useState(true);
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/pages/register/register.js:28]: Because at the beginning password is equal to the confirmedPassword ('' === '')

File: src/redux/user/user.reducer.js [hunk 28]
```diff
@@ -33,7 +40,37 @@ const userReducer = (state = initialState, action = {}) => {
   case SET_USER_LOADING:
     return {
       ...state,
-      userLoading: true
+      userLoading: action.payload
+    };
+  case STATE_RESET:
+    return {
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/redux/user/user.reducer.js:46]: Initial state has userData = null, if i set it to initialState it will logout basically. Here i just want everything reset, except userData

File: src/configs/index.js [hunk 29]
```diff
@@ -48,9 +48,54 @@ export const TIME_OPTIONS = {
   day: 'numeric'
 };
 
+
 export const LANGUAGES_LIST = [
   { lang: 'UA', value: 0 },
   { lang: 'EN', value: 1 }
+
+export const CONFIRM_EMAIL = [
```
Comment (Alan18081) [inline src/configs/index.js:56]: Please, create a file with translations inside the `translations` folder, this code should not be here, in config file

File: src/configs/index.js [hunk 30]
```diff
@@ -81,3 +126,105 @@ export const formRegExp = {
   deliveryType: "^(?=.{1,30}$)[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$",
   deliveryMethod: "^(?=.{1,30}$)[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$"
 };
+
+export const placeholders = {
```
Comment (Alan18081) [inline src/configs/index.js]: Please, create a file with translations inside the `translations` folder, this code should not be here, in config file

File: src/routes/routes.js [hunk 31]
```diff
@@ -54,6 +54,20 @@ const Routes = () => {
             exact
             render={({ match }) => <NewPassword token={match.params.token} />}
           />
+          <Route
```
Comment (Alan18081) [inline src/routes/routes.js:57]: Please, revert this change, it should not be part of this merge request

File: src/redux/user/user.sagas.js [hunk 32]
```diff
@@ -1,36 +1,172 @@
...
+  REGISTER_USER
+} from './user.types';
 import { setItems } from '../../utils/client';
+import { SHOW_AFTER } from '../../configs/index';
 
-export const loginUser = (payload) => {
+export const loginUser = (data) => {
   const query = ` 
-  mutation login($user: UserInput!) {
+  mutation mutation($user: LoginInput!){
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/redux/user/user.sagas.js:25]: reverted

File: src/redux/user/user.reducer.js [hunk 33]
```diff
@@ -33,7 +40,37 @@ const userReducer = (state = initialState, action = {}) => {
...
+      userRegistered: initialState.userRegistered
+    };
+  case SET_USER_RECOVERED:
+    return {
+      ...state,
+      userRecovered: action.payload,
+      error: false,
+      userLoading: false
+    };
+  case SET_PASSWORD_RESET:
```
[the remark this answers is not in the document]
Comment (INR4GE) [reply src/redux/user/user.reducer.js:61]: removed

File: src/redux/user/user.reducer.js (hunk 19 above)
Comment (INR4GE) [reply src/redux/user/user.reducer.js:72]: fixed

File: src/redux/user/user.sagas.js (hunk 20 above)
Comment (INR4GE) [reply src/redux/user/user.sagas.js:41]: moved

File: src/redux/user/user.sagas.js (hunk 21 above)
Comment (INR4GE) [reply src/redux/user/user.sagas.js:50]: moved

File: src/configs/index.js (hunk 30 above)
Comment (INR4GE) [reply src/configs/index.js]: moved

File: src/configs/index.js (hunk 29 above)
Comment (INR4GE) [reply src/configs/index.js:56]: moved

File: src/redux/user/user.sagas.js (hunk 25 above)
Comment (INR4GE) [reply src/redux/user/user.sagas.js:133]: renamed

File: src/configs/index.js (hunk 29 above)
Comment (INR4GE) [reply src/configs/index.js:56]: created

**Review by Alan18081 — APPROVED** [review]
